doc/overviews/helm-charts-release-process.md

```diff
@@ -0,0 +1,237 @@
+# Helm Chart Publishing Overview
+
+This document describes the automated process for publishing Kuadrant Operator Helm charts from development to [Artifact Hub](https://artifacthub.io/packages/helm/kuadrant/kuadrant-operator).
+
+## Overview
+
+The Kuadrant Operator Helm chart publishing process is fully automated and consists of five main phases:
```

Comment:
Maybe we could consider to split the Overview and Process Flow for 1. Development Only and 2. Releases